### PR TITLE
Vet clinic database: database performance audit

### PR DESCRIPTION
**Pull request by AlselwiAisha**
**In this repo:**

- Make sure that you have your database set up with the schema and data from your previous projects.
      - If you transitioned from 1.0 and you haven't complete the entire Ruby+Databases module, you do not have the database ready. In that case, please use the [schema.sql](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/images/schema.sql) to create tables and [data.sql](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/images/data.sql) to populate tables with the initial data.
- Run the following query to add an extra column to the owners table:

```sh
-- Add an email column to your owners table
ALTER TABLE owners ADD COLUMN email VARCHAR(120);
```

- Run the following statements to add data to your database (executing them might take a few minutes):

```sh
-- This will add 3.594.280 visits considering you have 10 animals, 4 vets, and it will use around ~87.000 timestamps (~4min approx.)
INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT * FROM (SELECT id FROM animals) animal_ids, (SELECT id FROM vets) vets_ids, generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') visit_timestamp;

-- This will add 2.500.000 owners with full_name = 'Owner <X>' and email = 'owner_<X>@email.com' (~2min approx.)
insert into owners (full_name, email) select 'Owner ' || generate_series(1,2500000), 'owner_' || generate_series(1,2500000) || '@mail.com';
```

- The following queries are taking too much time (1 sec = 1000ms can be considered as too much time for database query). Try them on your machine to confirm it:

```sh 
SELECT COUNT(*) FROM visits where animal_id = 4;
```
Week2_1
```sh
SELECT * FROM visits where vet_id = 2;
```
W2-2

```sh
SELECT * FROM owners where email = 'owner_18327@mail.com';
```
W2-3

### CODE REVIEW

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @AlselwiAisha  and @Izebiafe 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️schema is updated
✔️data is inserted
✔️screenshots added 
## Required Changes ♻️
- [ ] Creating indexes really worked according to the screenshots which are lower than 1000ms as we expected. Good job on that! However, according to the project requirements, there must be screenshots showing the results of how much it takes for each query before adding the improvements. Please add them too.
image
 
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AlselwiAisha** [comment]
**before adding indexes for the tables**:
```sh 
SELECT COUNT(*) FROM visits where animal_id = 4;
```


```sh
SELECT * FROM visits where vet_id = 2;
```



```sh
SELECT * FROM owners where email = 'owner_18327@mail.com';
```



**After adding indexes for the tables:**
```sh 
SELECT COUNT(*) FROM visits where animal_id = 4;
```


```sh
SELECT * FROM visits where vet_id = 2;
```


```sh
SELECT * FROM owners where email = 'owner_18327@mail.com';
```

File: README.md [hunk 1]
```diff
@@ -1,6 +1,5 @@
 # Vet Clinic Database
 
-
 # 📗 Table of Contents
```
Comment (julie-ify) [inline README.md]: - [ ] **[OPTIONAL]** It would be nice to fix the broken links on your readme headers, like deployment, run tests, etc

**Review by julie-ify — APPROVED** [review]
Hi @AlselwiAisha and @Izebiafe,

## Highlights!
- Good code organization and query optimizations ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**